backend/app/views/spree/admin/products/_form.html.erb

```diff
@@ -158,7 +158,7 @@
         <%= f.field_container :tax_category do %>
           <%= f.label :tax_category_id, Spree::TaxCategory.model_name.human %>
           <%= f.field_hint :tax_category, default_tax_category: @default_tax_category&.name %>
-          <%= f.collection_select(:tax_category_id, @tax_categories, :id, :name, { include_blank: t('spree.match_choices.none'), selected: @default_tax_category&.id }, { class: 'custom-select' }) %>
+          <%= f.collection_select(:tax_category_id, @tax_categories, :id, :name, { include_blank: t('spree.match_choices.none'), selected: f.object.tax_category_id || @default_tax_category&.id }, { class: 'custom-select' }) %>
```

Comment:
What about setting the default on `@product.tax_category` at the controller level?

Reply:
sure, that an option. Both solutions look rather equivalent to me. While the view gets simpler (we can entirely remove the `selected` specification here) the controller gains a `before_action` for both edit and new. 

I've updated the code, let me know what you think!

Reply:
I like this way! Let see if others have opinions as well.